pinot-controller/src/main/java/com/linkedin/pinot/controller/ControllerStarter.java

```diff
@@ -109,61 +110,68 @@ public void start() {
 
     Utils.logVersions();
 
-    component.getServers().add(Protocol.HTTP, Integer.parseInt(config.getControllerPort()));
-    component.getClients().add(Protocol.FILE);
-    component.getClients().add(Protocol.JAR);
-
-    final Context applicationContext = component.getContext().createChildContext();
-
-    LOGGER.info("Controller download url base: {}", config.generateVipUrl());
-    LOGGER.info("Injecting configuration and resource manager to the API context");
-    applicationContext.getAttributes().put(ControllerConf.class.toString(), config);
-    applicationContext.getAttributes().put(PinotHelixResourceManager.class.toString(), helixResourceManager);
-    MultiThreadedHttpConnectionManager connectionManager = new MultiThreadedHttpConnectionManager();
-    connectionManager.getParams().setConnectionTimeout(config.getServerAdminRequestTimeoutSeconds());
-    applicationContext.getAttributes().put(HttpConnectionManager.class.toString(), connectionManager);
-    applicationContext.getAttributes().put(Executor.class.toString(), executorService);
-
-    controllerRestApp.setContext(applicationContext);
-
-    component.getDefaultHost().attach(controllerRestApp);
-
+    // Set up controller metrics
     MetricsHelper.initializeMetrics(config.subset(METRICS_REGISTRY_NAME));
     MetricsHelper.registerMetricsRegistry(_metricsRegistry);
     final ControllerMetrics controllerMetrics = new ControllerMetrics(_metricsRegistry);
 
+    // Start all components
     try {
       LOGGER.info("Starting Pinot Helix resource manager and connecting to Zookeeper");
       helixResourceManager.start();
 
-      // Helix resource manager must be started in order to create PinotLLCRealtimeSegmentManager
-      PinotLLCRealtimeSegmentManager.create(helixResourceManager, config, controllerMetrics);
-      ValidationMetrics validationMetrics = new ValidationMetrics(_metricsRegistry);
-      _validationManager = new ValidationManager(validationMetrics, helixResourceManager, config,
-          PinotLLCRealtimeSegmentManager.getInstance());
-
+      LOGGER.info("Starting task resource manager");
       // Helix resource manager must be started in order to get TaskDriver
       TaskDriver taskDriver = new TaskDriver(helixResourceManager.getHelixZkManager());
       _helixTaskResourceManager = new PinotHelixTaskResourceManager(taskDriver);
+
+      LOGGER.info("Starting task manager");
```

Comment:
I suggest we make this configurable. In case it has some issues we can at least turn it off.